Source/Particles/Resampling/VelocityCoincidenceThinning.cpp

```diff
@@ -189,6 +195,10 @@ void VelocityCoincidenceThinning::operator() (WarpXParIter& pti, const int lev,
             // ones in the same momentum bin
             for (int i = cell_start; i < cell_stop; ++i)
             {
+                // skip merging if the momentum bin is 0
+                if (momentum_bin_number_data[sorted_indices_data[i]] == 0) {
+                    continue;
+                }
                 particles_in_bin += 1;
                 const auto part_idx = indices[sorted_indices_data[i]];
 
```

Comment:
In the `if` below at line 211, could an additional check be added `total_weight > m_max_w`? This would limit the merging of particles to the same maximum weight. Doing it this way would also remove the need for the checks above, setting the bin number to zero.

Reply:
That is a great idea! Basically instead of setting a maximum weight that particles cannot be merged beyond, it would set a target weight for each merging cluster. But I think it would be important to **not** include the particle that pushed the `total_weight` over the edge. So the `if` statement would be:
```
if (
    (i == cell_stop - 1)
    || (momentum_bin_number_data[sorted_indices_data[i]] != momentum_bin_number_data[sorted_indices_data[i + 1]])
    || (total_weight + w[indices[sorted_indices_data[i + 1]] > m_target_weight / 2.0)
) {
```
where the new `m_target_weight` parameter is introduced. Note the divide by 2 since half the total weight is given to each product particle.